tests/unit/llms/test_json_load.py

```diff
@@ -0,0 +1,58 @@
+import types
+import os
+import unittest.mock
+
+from openai.types.chat import ChatCompletionMessage
+from openai.types.chat.chat_completion import ChatCompletion, Choice
+
+from ragas.llms.base import llm_factory
+from ragas.llms.json_load import json_loader
+
+
+class MockOpenAI:
+    class MockCompletions:
+        def create(self, *args, **kwargs):
+            return ChatCompletion(
+                id="123456",
+                model="model",
+                object="chat.completion",
+                choices=[
+                    Choice(
+                        finish_reason="stop",
+                        index=0,
+                        message=ChatCompletionMessage(
+                            content='{"one": "two"},,, {"three": "four"}]',
+                            role="assistant",
+                        ),
+                    )
+                ],
+                created=123,
+            )
```

Comment:
should we mock openai or langchain_openai

Reply:
So I'm not sure why, but I can't seem to get the patch to catch a higher abstraction. Here's an example

```
class MockRagasLLM(BaseRagasLLM):
    def generate_text(
        self,
        prompt: PromptValue,
        n: int = 1,
        temperature: float = 1e-8,
        stop: t.Optional[t.List[str]] = None,
        callbacks: t.Optional[Callbacks] = None,
    ) -> LLMResult:
        return LLMResult(
            generations=[
                ChatGeneration(message=AIMessage('{"one": "two"},,, {"three": "four"}]'))
            ]
        )
```

then later on

```
        self.patches = [
...
            unittest.mock.patch(
                "ragas.llms.LangchainLLMWrapper",
                new=MockRagasLLM,
            ),
        ]
```

But again I can't get it to catch. I.e. I get a bad API key error from OpenAI, which shouldn't be happening because the patch should be catching the call to generate_text before it even gets to OpenAI.

Reply:
Committed it to the branch here: [9a76d41](https://github.com/explodinggradients/ragas/pull/659/commits/9a76d4106e5b6e8ddd62f117d1f6dfa93cc3e9d3)